src/service/daemon.js

```diff
@@ -41,10 +42,49 @@ const Service = GObject.registerClass({
         GLib.set_prgname('gsconnect');
         GLib.set_application_name('GSConnect');
 
+        // TODO: remove after a reasonable period of time
+        this._migrateConfiguration();
+
         // Command-line
         this._initOptions();
     }
 
+    _migrateConfiguration() {
+        if (Device.validateId(this.settings.get_string('id')))
+            return;
+
+        // Remove the old certificate, serving as the single source of truth
+        // for the device ID
+        try {
+            Gio.File.new_build_filenamev([Config.CONFIGDIR, 'certificate.pem'])
+                .delete(null);
+            Gio.File.new_build_filenamev([Config.CONFIGDIR, 'private.pem'])
+                .delete(null);
+        } catch {
+            // Silence errors
+        }
```

Comment:
This all looks sensible, but I wonder if we need some sort of notification to the user, when the migration process triggers? ("Hey, we had to re-generate your local device ID, so all of your devices need to be re-paired. Lusciously sorry!")

Reply:
Resolved in [641f5f9](https://github.com/GSConnect/gnome-shell-extension-gsconnect/pull/1911/commits/641f5f999db42e9d250c2acda12aa7f70700a3ed)